packages/agent/src/parsers.ts

```diff
@@ -259,12 +261,21 @@ export async function receiveOutOfBandInvitation(
     )
   )
 
-  const { connectionRecord, outOfBandRecord } = await agent.oob.receiveInvitation(invitation, {
-    reuseConnection: true,
-  })
+  let connectionRecord: ConnectionRecord | undefined, outOfBandRecord: OutOfBandRecord
+
+  try {
+    ;({ connectionRecord, outOfBandRecord } = await agent.oob.receiveInvitation(invitation, {
+      reuseConnection: true,
+    }))
 
-  // Assign connectionId so it can be used in the observables.
-  connectionId = connectionRecord?.id
+    // Assign connectionId so it can be used in the observables.
+    connectionId = connectionRecord?.id
+  } catch (error) {
+    return {
+      result: 'error',
+      message: 'Invitation has already been scanned.',
```

Comment:
There are quite some errors that could occur here. Can we somehow catch whether it really is the case that this invitation already has been scanned?

Maybe we can try to find an invitation by the id, and if a record is found we show the notification. That way we know for sure it is actually already scanned (instead of another error)